src/include/btree.i

```diff
@@ -1085,6 +1085,14 @@ __wt_page_can_evict(WT_SESSION_IMPL *session,
 		return (true);
 
 	/*
+	 * If the page is clean but has modifications that appear too
+	 * new to evict, skip it.
+	 */
+	if (!__wt_txn_visible_all(session, mod->rec_max_txn) ||
+	    !__wt_txn_visible_all(session, mod->update_txn))
```

Comment:
@daveh86 I suggested looking for different transaction IDs that might be more suitable here, but I don't think ``mod->update_txn`` is the one we are interested in. The ``update_txn`` is used as an advisory value, and shouldn't be involved in correctness checks (which we are pursuing here).

The expectation in the code is that either:
* The page is clean and ``rec_max_txn`` accurately reflects the ID of the most recent update on the page (which needs to be kept around as long as the oldest reader was started before the update).
* The page is dirty, and the state of the update lists are checked during reconciliation.

Can you reproduce this failure reliably? If so, could you capture the assertion in a debugger? We would want to be able to see the fully populated page-modify structure, so we may need to restructure the code a bit to find the problem before cleaning up content.